syntax = "proto3";

package grpc_service;

service TestService {
  rpc SayHello (HelloRequest) returns (HelloResponse) {}
  rpc HealthCheck (HealthCheckRequest) returns (HealthCheckResponse) {}
  rpc GetMetaData (MetaRequest) returns (MetaResponse) {}
}

message HelloRequest {
  string name = 1;
}

message HelloResponse {
  string message = 1;
}

message HealthCheckRequest {
  string service = 1;
}

message HealthCheckResponse {
  string status = 1;
}

message MetaRequest {
  string key = 1;
  string value = 2;
}

message MetaResponse {
  repeated string data = 1;
}
